packages/material-ui/src/Radio/RadioButtonIcon.js

```diff
@@ -3,47 +3,49 @@ import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import RadioButtonUncheckedIcon from '../internal/svg-icons/RadioButtonUnchecked';
 import RadioButtonCheckedIcon from '../internal/svg-icons/RadioButtonChecked';
-import withStyles from '../styles/withStyles';
+import experimentalStyled from '../styles/experimentalStyled';
 
-export const styles = (theme) => ({
-  root: {
-    position: 'relative',
-    display: 'flex',
-    '&$checked $dot': {
-      transform: 'scale(1)',
-      transition: theme.transitions.create('transform', {
-        easing: theme.transitions.easing.easeOut,
-        duration: theme.transitions.duration.shortest,
-      }),
-    },
-  },
-  checked: {},
-  background: {
-    // Scale applied to prevent dot misalignment in Safari
+const RadioButtonIconRoot = experimentalStyled('span')(({ theme }) => ({
+  position: 'relative',
+  display: 'flex',
+  '&.Mui-checked .MuiRadioButtonIcon-dot': {
     transform: 'scale(1)',
-  },
-  dot: {
-    left: 0,
-    position: 'absolute',
-    transform: 'scale(0)',
     transition: theme.transitions.create('transform', {
-      easing: theme.transitions.easing.easeIn,
+      easing: theme.transitions.easing.easeOut,
       duration: theme.transitions.duration.shortest,
     }),
   },
+}));
+
+const RadioButtonIconBackground = experimentalStyled(RadioButtonUncheckedIcon)({
+  // Scale applied to prevent dot misalignment in Safari
+  transform: 'scale(1)',
 });
 
+const RadioButtonIconDot = experimentalStyled(RadioButtonCheckedIcon)(({ theme, styleProps }) => ({
+  left: 0,
+  position: 'absolute',
+  transform: 'scale(0)',
+  transition: theme.transitions.create('transform', {
+    easing: theme.transitions.easing.easeIn,
+    duration: theme.transitions.duration.shortest,
+  }),
+}));
+
 /**
  * @ignore - internal component.
  */
 function RadioButtonIcon(props) {
-  const { checked, classes, fontSize } = props;
+  const { checked, classes = {}, fontSize } = props;
 
   return (
-    <span className={clsx(classes.root, { [classes.checked]: checked })}>
-      <RadioButtonUncheckedIcon fontSize={fontSize} className={classes.background} />
-      <RadioButtonCheckedIcon fontSize={fontSize} className={classes.dot} />
-    </span>
+    <RadioButtonIconRoot className={clsx(classes.root, { 'Mui-checked': checked })}>
```

Comment:
Does it need to use `.Mui-checked`? It seems to be a private component that could leverage `styleProps`.

Reply:
Wasn't sure whether is fine to change the specificity... In that case, we can add the style in the slot where it is required and we won't bump the specificity at all in the styles definition

Reply:
Updated to depend on `styleProps`

Reply:
Good point, maybe it will conflict with inheritance

Reply:
I couldn't spot any issues, updated to depend on `styleProps`